app/models/post.rb

```diff
@@ -1,5 +1,26 @@
+# class Post < ApplicationRecord
+#   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+#   has_many :comments
+#   has_many :likes
+# end
+
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  has_many :likes
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
   has_many :comments
+  has_many :likes
+
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comment_counter, :integer, default: 0
+  attribute :like_counter, :integer, default: 0
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] instead of using `update` I would suggest you to use `increment!` method as this method  directly modifies the attribute in the database without reloading the object while `update` method triggers a database update and reloads the object. Example snippet:

```rb
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```